From 0312477d90512f341a7eb66665ff89bb76b1d889 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <lucianafujii@collabora.co.uk>
Date: Wed, 19 Oct 2011 15:27:58 -0400
Subject: [PATCH 104/119] pvrvideosink: Check buffers allocation

Don't add buffers to available queue if allocation has failed.
---
 sys/pvr2d/gstpvr.c           |    2 +-
 sys/pvr2d/gstpvrbufferpool.c |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/sys/pvr2d/gstpvr.c b/sys/pvr2d/gstpvr.c
index e440e95..0112d1e 100644
--- a/sys/pvr2d/gstpvr.c
+++ b/sys/pvr2d/gstpvr.c
@@ -81,5 +81,5 @@ gst_ducati_alloc_2d (gint width, gint height, guint * sz)
 GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
     GST_VERSION_MINOR,
     "pvr",
-    "Hardware accellerated codecs for OMAP4",
+    "Pvr2d based plugin",
     plugin_init, VERSION, "LGPL", "GStreamer", "http://gstreamer.net/")
diff --git a/sys/pvr2d/gstpvrbufferpool.c b/sys/pvr2d/gstpvrbufferpool.c
index c8ad2c3..5e46572 100644
--- a/sys/pvr2d/gstpvrbufferpool.c
+++ b/sys/pvr2d/gstpvrbufferpool.c
@@ -193,6 +193,7 @@ gst_pvr_bufferpool_new (GstElement * element, GstCaps * caps, gint num_buffers,
   /* now, map the buffers */
   for (i = 0; i < num_buffers; i++) {
     self->buffers[i] = gst_ducati_buffer_new (self, i);
+    g_return_val_if_fail (self->buffers[i] != NULL, NULL);
     self->num_live_buffers++;
     g_async_queue_push (self->avail_buffers, self->buffers[i]);
   }
-- 
1.7.5.4

